media/docs/cpp/cute/02_layout_algebra.md

```diff
@@ -159,11 +169,69 @@ As you may have noticed, we can only divide shapes by certain values and get a s
 * `(1,2,2,8) %  2 => (1,2,1,1)`
 * `(1,2,2,8) % 16 => (1,2,2,4)`
 
+Essentially, this operation finds a layout shape that remains compatible with `B`, while maintaining our new stride.
+
 Again, this operation must satisfy a **shape divisibility condition** to yield a sensible result and is statically checked in CuTe when possible.
 
-From the above examples, we can construct the composition `(3,6,2,8):(w,x,y,z) o 16:9 = (1,2,2,4):(3*w,3*x,y,z)`.
+Returning to our example `(6,2):(x, y) o (2, 2)`, we see that "modding out" to find the required shape results in `(6, 2) % 2 = (2, 1)`. This results in a final composition `(6,2):(x, y) o (2, 2) = (2, 1): (2*x, y)`.
+
+---
+
+#### Example 1 -- Worked Example of Calculating a Composition
+
+We provide a more complex example of composition, where both operand layouts are multi-modal to illustrate the concepts introduced above.
+
+```
+Functional composition, R := A o B
+R(c) := (A o B)(c) := A(B(c))
+
+Example
+A = (6,2):(8,2)
+B = (4,3):(3,1)
+
+1. Using the left-distributive and concatenation properties for layouts we write the composition as,
 
-#### Example 1 -- Reshape a layout into a matrix
+R = A o B
+  = (6,2):(8,2) o (4,3):(3,1)
+  = ((6,2):(8,2) o 4:3, (6,2):(8,2) o 3:1)
+
+---
+1. Beginning with (6,2):(8,2) o 4:3
+
+- Calculate the shape of the new strided view of the composition
+
+(6, 2) / 3 = (2, 2)
+
+- This implies a stride of
+
+(8*3, 2) = (24, 2)
+
+- Calculate the shape of the composition
+
+(6, 2) % 4 = (2, 2)
+
+- We note that the operation above must respect the two-dimensional nature of the stride over A, so (4, 1) is not acceptable.
```

Comment:
This is the incorrect intuition and is misleading.

First, a version of `A` that is strided by `d` is computed. Then, the first `s` elements *of that strided layout* are "kept". The first `s` elements of the original `A` play no role and there is no "acceptability" or "two-dimensional" nature to consider.